zookeeper/src/main/java/com/linecorp/armeria/client/zookeeper/ZooKeeperEndpointGroup.java

```diff
@@ -54,81 +55,97 @@
      * Returns a new {@link ZooKeeperEndpointGroup} that retrieves the {@link Endpoint} list from
      * the ZNode at the specified connection string and path. A new ZooKeeper client will be created internally.
      * The ZooKeeper client will be destroyed when the returned {@link ZooKeeperEndpointGroup} is closed.
+     *
+     * @param zkConnectionStr the ZooKeeper connection string
+     * @param zNodePath the ZooKeeper node to register
+     * @param spec the {@link ZookeeperDiscoverySpec} to find and decode the registered instances
      */
-    public static ZooKeeperEndpointGroup of(String zkConnectionStr, String zNodePath) {
-        return builder(zkConnectionStr, zNodePath).build();
+    public static ZooKeeperEndpointGroup of(String zkConnectionStr, String zNodePath,
+                                            ZookeeperDiscoverySpec spec) {
```

Comment:
It's time to update the documentation. 😉
https://line.github.io/armeria/docs/advanced-zookeeper

Reply:
Let me make a separate PR for that as I commented in the commit message.
I want to implement the specs for ServerSet first and then document.

Reply:
SGTM 👍